webrender/res/ps_angle_gradient.fs.glsl

```diff
@@ -5,11 +5,14 @@
 uniform sampler2D sGradients;
 
 void main(void) {
+    // Normalized offset of this vertex within the gradient, before clamp/repeat.
+    float offset = dot(vPos - vStartPoint, vDir) / dot(vDir, vDir);
```

Comment:
Let's have `vDir` adjusted in the vertex shader before interpolation in order to avoid the computation here:
```glsl
vec2 dir = end_point - start_point;
vDir = dir / dot(dir, dir);
```